Sources/Amplitude/Storages/PersistentStorage.swift

```diff
@@ -362,23 +378,110 @@ extension PersistentStorage {
+            updatedFile = fileWithoutTemp.appendFileNameSuffix(suffix: suffix)
+        }
         do {
-            try fileManager.moveItem(at: file, to: fileWithoutTemp)
+            try fileManager.moveItem(at: file, to: updatedFile)
         } catch {
+            amplitude?.diagonostics.addErrorLog(error.localizedDescription)
             amplitude?.logger?.error(message: "Unable to rename file: \(file.path)")
         }
+    }
 
-        let currentFileIndex: Int = (getCurrentEventFileIndex() ?? 0) + 1
-        userDefaults?.set(currentFileIndex, forKey: eventsFileKey)
+    private func handleV1Files() {
+        syncQueue.sync {
+            let currentStorageVersion = (userDefaults?.object(forKey: self.storageVersionKey) as? Int) ?? 0
+            if currentStorageVersion > 1 {
+                return
+            }
+            let allFiles = self.getEventFiles(includeUnfinished: true)
+            for file in allFiles {
+                do {
+                    let content = try String(contentsOf: file, encoding: .utf8)
+                    if content.hasSuffix(PersistentStorage.DELMITER) {
+                        break // already handled and in v2 format
+                    }
+
+                    let normalizedContent = "[\(content.trimmingCharacters(in: ["[", ",", "]"]))]"
+                    let events = BaseEvent.fromArrayString(jsonString: normalizedContent)
+                    if events != nil {
+                        migrateFile(file: file, events: events!)
+                    }
+                    if file.pathExtension != "" {
+                        finish(file: file)
+                    }
+                } catch {
+                    amplitude?.diagonostics.addErrorLog("Error migrating file: \(file.path) for \(error.localizedDescription)")
+                    amplitude?.logger?.error(message: error.localizedDescription)
+                }
+            }
+            userDefaults?.setValue(2, forKey: self.storageVersionKey)
+        }
+    }
+
+    private func migrateFile(file: URL, events: [BaseEvent]) {
+        guard fileManager.fileExists(atPath: file.path) == true else {
+            amplitude?.diagonostics.addErrorLog("File to migrate not exists any more : \(file.path)")
+            return
+        }
+
+        do {
+            let jsonString = events.map { $0.toString().replacingOccurrences(of: PersistentStorage.DELMITER, with: "")  }.joined(separator: PersistentStorage.DELMITER)
+            let finalString = "\(jsonString)\(PersistentStorage.DELMITER)"
+            try finalString.write(to: file, atomically: true, encoding: .utf8)
+        } catch {
+            amplitude?.diagonostics.addErrorLog(error.localizedDescription)
+            amplitude?.logger?.error(message: error.localizedDescription)
+        }
+    }
+
+    private func readV2File(content: String) -> String {
+        var events: [BaseEvent] = [BaseEvent]()
+        if content.hasSuffix(PersistentStorage.DELMITER) {
```

Comment:
Do we need to check this? Worried about the possible cases where we may drop data if the delimiter failed to write. Again, having the file version in the file name will help us be certain that its a v2 file instead of looking for a character in the file contents.

Reply:
This was removed